flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/stream/StreamExecTemporalJoin.scala

```diff
@@ -84,6 +110,261 @@ class StreamExecTemporalJoin(
+      textualRepresentation,
+      leftType.getFieldCount,
+      joinInfo,
+      rexBuilder)
+
+    val nonEquiJoinRex: RexNode = joinInfo.getRemaining(rexBuilder)
+    val remainingNonEquiJoinPredicates = temporalJoinConditionExtractor.apply(nonEquiJoinRex)
+
+    val (leftTimeAttributeInputRef, rightRowTimeAttributeInputRef: Option[Int]) =
+      if (TemporalJoinUtil.isRowTimeJoin(rexBuilder, joinInfo)) {
+        checkState(temporalJoinConditionExtractor.leftTimeAttribute.isDefined &&
+          temporalJoinConditionExtractor.rightPrimaryKey.isDefined,
+          "Missing %s in Event-Time temporal join condition", TEMPORAL_JOIN_CONDITION)
+
+        val leftTimeAttributeInputRef = extractInputRef(
+          temporalJoinConditionExtractor.leftTimeAttribute.get, textualRepresentation)
+        val rightTimeAttributeInputRef = extractInputRef(
+          temporalJoinConditionExtractor.rightTimeAttribute.get, textualRepresentation)
+        val rightInputRef = rightTimeAttributeInputRef - leftType.getFieldCount
+
+        (leftTimeAttributeInputRef, Some(rightInputRef))
+      } else {
+        val leftTimeAttributeInputRef = extractInputRef(
+          temporalJoinConditionExtractor.leftTimeAttribute.get, textualRepresentation)
+        // right time attribute defined in temporal join condition iff in Event time join
+        (leftTimeAttributeInputRef, None)
+      }
+
+    new StreamExecTemporalJoinToCoProcessTranslator(
+      textualRepresentation,
+      config,
+      returnType,
+      leftType,
+      rightType,
+      joinInfo,
+      rexBuilder,
+      leftTimeAttributeInputRef,
+      rightRowTimeAttributeInputRef,
+      remainingNonEquiJoinPredicates)
+  }
+
+  private def extractInputRef(rexNode: RexNode, textualRepresentation: String): Int = {
+    val inputReferenceVisitor = new InputRefVisitor
+    rexNode.accept(inputReferenceVisitor)
+    checkState(
+      inputReferenceVisitor.getFields.length == 1,
+      "Failed to find input reference in [%s]",
+      textualRepresentation)
+    inputReferenceVisitor.getFields.head
+  }
+
+  private class TemporalJoinConditionExtractor(
+    textualRepresentation: String,
+    rightKeysStartingOffset: Int,
+    joinInfo: JoinInfo,
+    rexBuilder: RexBuilder)
+
+    extends RexShuttle {
+
+    var leftTimeAttribute: Option[RexNode] = None
+
+    var rightTimeAttribute: Option[RexNode] = None
+
+    var rightPrimaryKey: Option[Array[RexNode]] = None
+
+    override def visitCall(call: RexCall): RexNode = {
+      if (call.getOperator != TEMPORAL_JOIN_CONDITION) {
+        return super.visitCall(call)
+      }
+
+      if (TemporalJoinUtil.isRowTimeTemporalJoinConditionCall(call)) {
+        leftTimeAttribute = Some(call.getOperands.get(0))
+        rightTimeAttribute = Some(call.getOperands.get(1))
+        rightPrimaryKey = Some(extractPrimaryKeyArray(call.getOperands.get(4)))
+      } else {
+        leftTimeAttribute = Some(call.getOperands.get(0))
```

Comment:
It seems that we never check primary key on the right side when it is processing time temporal join. 
I think currently, we only support primary key == join key. 

And could you add such a unit test for this exception?

Reply:
I think the  primary key contained in join key is not necessary for processing time temporal join